lib/matplotlib/backends/backend_svg.py

```diff
@@ -1235,7 +1235,11 @@ def _print_svg(self, filename, svgwriter, fh_to_close=None, detach=False,
             renderer.finalize()
         finally:
             if detach:
-                svgwriter.detach()
+                if six.PY3:
+                    svgwriter.detach()
+                else:
+                    svgwriter.reset()
+                    svgwriter.stream = io.BytesIO()
```

Comment:
I think you are doing it not in a right way. Just `RendererSVG.writer` should be deleted in `RendererSVG.finalize`.

Closes #6870